File: api/function/MatchFunctions.js
const db = require('../config/db');/* 
const mongoose = require('mongoose');
const Match = require('../models/Match'); */

let MatchFunctions = {
    'insertOneMatch': (season, match, callback) => {
        db.find({
            database: season,
            collection: 'matches',
            query: {
                'date': match['date'],
                'homeTeam': match['homeTeam'],
                'awayTeam': match['awayTeam']
            },
            fields: {}
        }, function (data) {
            if (data.length == 0) {
                db.insertOne({
                    database: season,
                    collection: 'matches',
                    query: {},
                    fields: {}
                }, match, function () {
                    callback(true);
                })
            } else {
                callback(false);
            }
        });
    },
    'updateLastDate': (season, callback) => {
        let date = new Date(new Date().toUTCString());
        db.updateOne({
            database: season,
            collection: 'matches',
            query: {
                type: "date"
            },
            fields: {
                _id: 0,
                __v: 0
            }
        }, {
                $set: {
                    date: date
                }
            }, function (data) {
                callback(date);
            })
    },
    'getLastUpdated': (season, callback) => {
        db.findOne({
            database: season,
            collection: 'matches',
            query: {
                type: "date"
            },
            fields: {
                _id: 0,
                __v: 0
            }
        }, function (data) {
            callback(data);
        })
    },
    'findAllMatches': (season, callback) => {
        db.find({
            database: season,
            collection: 'matches',
            query: {},
            fields: {
                _id: 0,
                __v: 0
            }
        }, function (data) {
            callback(data);
        })
    },
    'fineOneTeamMatches': (teamName, callback) => {
        db.find('matches', 'matches', {
            $or: [{
                'homeTeam': teamName
            }, {
                'awayTeam': teamName
            }]
        }, {}, function (data) {
            callback(data);
        });
    },
    'findOneMatch': (teamA, teamB, callback) => {
        db.find('matches', 'matches', {
            $or: [{
                'homeTeam': teamA,
                'awayTeam': teamB
            },
            {
                'homeTeam': teamB,
                'awayTeam': teamA
            }
            ]
        }, {
                _id: 0,
                __v: 0
            }, function (data) {
                callback(data);
            });
    }
}

module.exports = MatchFunctions;

File: api/functions/MatchFunctions.py
import json
from .db import find, findOne, insertOne, updateOne
from bson.json_util import dumps
import sys
from datetime import datetime


def getOneMatch(homeTeam, awayTeam):
    db_params = {
        'database': 'season_5_2018',
        'collection': 'matches',
        'query': {
            '$or': [
                {'team': homeTeam},
                {'team': awayTeam}
            ]
        }
    }
    match = findOne(db_params)

    return match


def insertOneMatch(season, match):
    db_params = {
        'database': season,
        'collection': 'matches',
        'query': {
            'date': match['date'],
            'homeTeam': match['homeTeam'],
            'awayTeam': match['awayTeam']
        }
    }
    doc = findOne(db_params)
    if doc is None:
        doc = insertOne(db_params, match)
        print("Inserted document " + str(doc), file=sys.stderr)
        return True
    else:
        print("Document existed", file=sys.stderr)
        return False


def updateLastDate(season):
    db_params = {
        'database': season,
        'collection': 'matches',
        'query': {
            'type': 'date'
        }
    }
    date_string = str(datetime.utcnow())
    print("DATE :::::::::", file=sys.stderr)
    print(date_string, file=sys.stderr)
    doc = updateOne(db_params, {
        '$set': {
            'date': date_string
        }
    })
    if doc is not None:
        return date_string
    else:
        print('COULD NOT UPDATE DATE ERROR', file=sys.stderr)
        return False


def getLastDate(season):
    db_params = {
        'database': season,
        'collection': 'matches',
        'query': {
            'type': 'date'
        }
    }
    doc = findOne(db_params)
    return doc
